project/Build.scala

```diff
@@ -268,6 +270,36 @@ object Build {
 
     // enable verbose exception messages for JUnit
     (Test / testOptions) += Tests.Argument(TestFrameworks.JUnit, "-a", "-v", "-s"),
+
+    // Configuration to publish build scans to develocity.scala-lang.org
+    develocityConfiguration := {
+      val isInsideCI = insideCI.value
+      val previousConfig = develocityConfiguration.value
+
+      val tags = Set(if (isInsideCI) "CI" else "Local")
+      val workflowUrl =
+        for {
+          serverUrl <- Properties.envOrNone("GITHUB_SERVER_URL")
+          repository <- Properties.envOrNone("GITHUB_REPOSITORY")
+          runId <- Properties.envOrNone("GITHUB_RUN_ID")
+        } yield "workflow" -> url(s"$serverUrl/$repository/actions/runs/$runId")
+      val githubValues = Seq("GITHUB_REPOSITORY", "GITHUB_SHA", "GITHUB_WORKFLOW")
+        .flatMap(key => Properties.envOrNone(key).map(key -> _))
+
+      val previousBuildScan = previousConfig.buildScan
+      previousConfig
+        .withProjectId(ProjectId("scala3"))
+        .withServer(previousConfig.server.withUrl(Some(url("https://develocity.scala-lang.org"))))
+        .withBuildScan(
+          previousBuildScan
+            .withPublishing(Publishing.onlyIf(_.authenticated))
+            .withBackgroundUpload(!isInsideCI)
+            .withTags(previousBuildScan.tags ++ tags)
```

Comment:
There is a `def tag(tag: String): BuildScan` that you may like better. Just making you aware. Similarly, we have `link(link: String, url: URL)` and `value(key: String, value: String)`, both returning a `BuildScan` instance. No need to make any change of course.